domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientTest.java

```diff
@@ -709,5 +715,96 @@ public void getTotalRecordsTest() {
 
     assertEquals((Integer) 25, PostgresClient.getTotalRecords(5, 20, 20, 10));
   }
-
+  
+  @Test
+  public void testGetById() {
+    String table = "a";
+    String id = UUID.randomUUID().toString();
+    PostgresClient pc = spy(PostgresClient.testClient());
+    
+    // mock empty query result
+    @SuppressWarnings("unchecked")
+    RowSet<Row> mockRowSet = mock(RowSet.class);
+    when(mockRowSet.size()).thenReturn(0);
+    @SuppressWarnings("unchecked")
+    PreparedQuery<RowSet<Row>> mockPreparedQuery = mock(PreparedQuery.class);
+    doAnswer(AdditionalAnswers.answerVoid(
+        (Tuple tuple, Handler<AsyncResult<RowSet<Row>>> handler)
+        -> handler.handle(Future.succeededFuture(mockRowSet))))
+        .when(mockPreparedQuery).execute(any(Tuple.class), any());
+    PgConnection mockPgConnection = mock(PgConnection.class);
+    when(mockPgConnection.preparedQuery(anyString())).thenReturn(mockPreparedQuery);
+    PgPool mockPgPool = mock(PgPool.class);
+    doAnswer(AdditionalAnswers.answerVoid((Handler<AsyncResult<PgConnection>> handler)
+        -> handler.handle(Future.succeededFuture(mockPgConnection))))
+        .when(mockPgPool).getConnection(any());
+    when(pc.getClient()).thenReturn(mockPgPool);
+    SQLConnection mockSQLConnection = new SQLConnection(mockPgConnection, null, null);
+    AsyncResult<SQLConnection> mockConn = Future.succeededFuture(mockSQLConnection);
+    // tests
+    pc.getByIdAsString(table, id, ar -> assertTrue(ar.succeeded()));
```

Comment:
If the method calls the callback then `ar -> assertTrue(ar.succeeded())` will run.
If the method doesn't complete and doesn't call the callback then `ar -> assertTrue(ar.succeeded())` will NOT run but the unit test will succeed and does NOT report this incomplete execution.
Please use https://vertx.io/docs/vertx-unit/java/ or https://vertx.io/docs/vertx-junit5/java/ that provide test methods for such tests, for example testContext.asyncAssertSuccess() or vertxTestContext.succeeding().